lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1986,6 +1986,12 @@ def cmd_set_help
   # Sets a name to a value in a context aware environment.
   #
   def cmd_set(*args)
```

Comment:
This is another hack that should be avoided by refactoring payload selection and moving this code to an auxiliary module.